src/replication/adapter/adapter.go

```diff
@@ -31,9 +31,13 @@ const (
 )
 
 var registry = map[model.RegistryType]Factory{}
+var adapterInfoMap = map[model.RegistryType]*model.AdapterPattern{}
```

Comment:
Do we need the map? How about to call the AdapterPattern method of registered factory to get the pattern when needed?